assets/js/theme/global/maintenanceMode.js

```diff
@@ -7,17 +7,55 @@ import $ from 'jquery';
 export default function (maintenanceMode = {}) {
     const header = maintenanceMode.header;
     const notice = maintenanceMode.notice;
+    const password = maintenanceMode.password;
+    const securePath = maintenanceMode.secure_path;
 
-    if (!(header && notice)) {
+    // Return if header & notice is null or if securePath is not defined (means inside theme editor)
+    if (!(header && notice) || !securePath) {
         return;
     }
 
-    const $element = $('<div>', {
-        id: 'maintenance-notice',
-        class: 'maintenanceNotice',
-    });
+    if (password) {
+        const $element = $('<div>', {
+            class: 'adminBar',
+        });
 
-    $element.html(`<p class="maintenanceNotice-header">${header}</p>${notice}`);
+        $element.html(`<div class="adminBar-logo">
+            <a href="${securePath}/manage/dashboard"><svg><use xlink:href="#logo-small"></use></svg></a></div>
+            <div class="adminBar-content">
+                <a href="${securePath}/manage/theme-editor" target="_blank">Customize Theme</a>
+                <div class="adminBar-private">
+                    <span>Your storefront is private.</span>
+                    <span class="preview">Share your site with preview code: ${password}</span>
+                </div>
+            </div>`);
 
-    $('body').append($element);
+        $('body').addClass('hasAdminBar');
+        $('body').prepend($element);
+
+        $('.adminBar-content a').click(attachThemeEditorParams);
+
+        $(window).scroll(() => {
```

Comment:
Oh crap, I was confused. I thought this was a template file. Nvm, everything is fine